src/azul/service/responseobjects/hca_response_v5.py

```diff
@@ -805,8 +815,8 @@ def create_facet(contents):
         def choose_entry(_term):
             if 'key_as_string' in _term:
                 return _term['key_as_string']
-            elif not isinstance(_term['key'], str):
-                return str(_term['key'])
+            elif isinstance(_term['key'], bool):
```

Comment:
This changes what happens if `_term['key']` is an int. Before it would be converted to a string. Now it remains an int. Is that your intention?

Reply:
Modified function to handle all cases, now will return a string or None. Boolean values converted and lowercased to match previous behavior. Numeric values converted to string.